providers/flagd/pkg/service/grpc/service.go

```diff
@@ -0,0 +1,175 @@
+package grpc_service
+
+import (
+	ctx "context"
+	"errors"
+
+	models "github.com/open-feature/flagd/pkg/model"
+	models2 "github.com/open-feature/golang-sdk-contrib/providers/flagd/pkg/model"
+	"github.com/open-feature/golang-sdk-contrib/providers/flagd/pkg/service"
+	of "github.com/open-feature/golang-sdk/pkg/openfeature"
+	log "github.com/sirupsen/logrus"
+	schemaV1 "go.buf.build/grpc/go/open-feature/flagd/schema/v1"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type GRPCServiceConfiguration struct {
+	Port int32
+	Host string
+}
+
+type GRPCService struct {
+	Client IGRPCClient
+}
+
+func (s *GRPCService) ResolveBoolean(flagKey string, context of.EvaluationContext, options ...service.IServiceOption) (*schemaV1.ResolveBooleanResponse, error) {
+	client := s.Client.GetInstance()
```

Comment:
There's a fair amount of repeated code in these funcs, can this be simplified?

Reply:
I believe that without access to generics trying to simplify this code would be difficult, the response type needs to match that of the http service and as a result the response structures would need to be constructed outside of the helpers - which would leave the code in a reasonably similar position to how it is at the moment.
Maybe this can still be done once the code base is moved to 1.18